### ISSUE TITLE
Amount operator + should return null if any amount is null

### ISSUE THREAD

**Issue by uncle-buzz** [petermorlion/RedStar.Amounts#5]
I'm not agree with your + operator in amount : 
```
public static Amount operator +(Amount left, Amount right)
        {
            if ((left == null) && (right == null)) return null;
            left = left ?? Zero((right != null) ? right._unit : Unit.None);
            right = right ?? Zero(left.Unit);
            return new Amount(left._value + right.ConvertedTo(left._unit)._value, left._unit);
        }
```
To my mind, something + null == null + something == null.
In your method, null + null == null, but something + null == null + something == Zero (Unit.None) which is a real amount (compatible to percent or decibel...).

It doesn't work like operators * and / who return null if any amount is null.

You could introduce something like Double.Nan in your class aount instead of null value, because you could create an amount with Double.NaN as value which is something like null.

**Comment by petermorlion** [comment]
Thanks, I'll look into this.

**Comment by petermorlion** [comment]
I can agree on this. The .NET Framework does the same thing, i.e. `var i = 8 + null` will product `null` for `i`. But then maybe I should also change the * and / operators.

**Comment by uncle-buzz** [comment]
For * and / operator, you already coded this behavior : 
```
public static Amount operator *(Amount left, Amount right)
        {
            if (ReferenceEquals(left, null))
                return null;
            else if (ReferenceEquals(right, null))
                return null;
            else
                return new Amount(left._value * right._value, left._unit * right._unit);
        }
```

**Comment by petermorlion** [comment]
Yeah, just noticed it :)

**Comment by petermorlion** [comment]
Fixed in release 4.0.0
